cmd/tnf/generate/catalog/catalog.go

```diff
@@ -116,6 +121,119 @@ func emitTextFromFile(filename string) error {
 	return nil
 }
 
+// it extracts the suite name and test name from an url such as
+// http://test-network-function.com/tests-case/SuitName/TestName
+
+func getSuiteAndTestFromUrl(url string, base_domain string) []string {
+	result := strings.Split(url, base_domain)
+	// len 2, because returns [0] before base_domain and [1] after base_domain
+	if len(result) > 2 {
+		fmt.Fprintf(os.Stderr, "Identifier Url not valid\n")
+		return nil
+	}
+
+	suite_test := strings.Split(result[1], "/")
+	return suite_test
+}
+
+// it extracts the test name and test name from an url such as
+// http://test-network-function.com/tests/TestName
+
+func getTestFromUrl(url string, base_domain string) string {
+	result := strings.Split(url, base_domain)
+	// len 2, because returns [0] before base_domain and [1] after base_domain
+	if len(result) > 2 {
+		fmt.Fprintf(os.Stderr, "Identifier Url not valid\n")
+		return ""
+	}
+	test_name := result[1]
+	return test_name
+}
+
+// it turns a list of identifiers
+// { url, version  }
+// and takes urls like http://test-network-function.com/testcases/SuiteName/TestName
+// to build a more structured catalogue
+// {
+//     suiteNameA: [testName [identifiers], testName2 [identifiers] ]
+//     suiteNameB: [testName3 [identifiers], testName4 [identifiers] ]
+// }
+func createPrintableCatalogFromIdentifiers(keys []claim.Identifier) map[string][]CatalogElement {
+	base_domain := identifiers.TestIDBaseDomain + "/"
+
+	catalog := make(map[string][]CatalogElement)
+	// we need the list of suite's names
+	var element CatalogElement
+
+	for _, i := range keys {
+		suite_test := getSuiteAndTestFromUrl(i.Url, base_domain)
```

Comment:
This is similar to XformToGinkgoItIdentifier(). Why not use that or at least keep the base domain related logic inside the identifiers package (instead of exporting the actual domain)?

Reply:
I fully agree, it would be nice to export from identifiers packages this kind of functionality. The problem with XformToGinkgoItIdentifier() is that only returns the TestName, but you cannot get Suite Name. So you cannot show the catalogue classified in the way of
SuiteName
|
|- TestName
|- TestNAme
SuiteName
|
|- Testname
we would need something opossite to formTestURL:

https://github.com/test-network-function/test-network-function/blob/main/test-network-function/identifiers/identifiers.go#L55-L58


But the other way around. So from an url you can de-compose and get the suite and test names.

Reply:
@jun-rh  do you think we could add these methods to the identifier's packages?
 * pkg/tnf/identifiers/identifiers.go  has no function that allows to get the TestName from the identifier url. Here there is no SuitNAme, or SuitName is just Test Case Building Block
 * test-network-functions/identifiers/identifiers.go has the XformToGinkgoItIdentifier that we can use to get the TestName, but we need other similar to get the SuiteName.
 Do you agree to add these changes? Can we use this same pull request?

Reply:
Done it. Now the functions that helps you get SuiteName and TestName from identifiers.urls moved to each corresponding package.
A little bit tricky because each package works differently about managing Identifiers. 
I have seen one of the packages has some tests. So I have added a new one for the new added function.